dtfs-central-api/src/v1/controllers/tfm/deal/tfm-get-deals.controller.js

```diff
@@ -2,63 +2,32 @@ const moment = require('moment');
 const escapeStringRegexp = require('escape-string-regexp');
 const db = require('../../../../drivers/db-client');
 const CONSTANTS = require('../../../../constants');
-const getObjectPropertyValueFromStringPath = require('../../../../utils/getObjectPropertyValueFromStringPath');
-const mapDataModel = require('../../../../mapping/mapDataModel');
-const setEmptyIfNull = require('../../../../utils/setEmptyIfNull');
 const {
   isTimestampField,
   dayStartAndEndTimestamps,
 } = require('./tfm-get-deals-date-helpers');
+const { getBSSProperty } = require('../../../../mapping/mapDataModel');
 
-const sortDeals = (deals, sortBy) =>
-  deals.sort((xDeal, yDeal) => {
-    const xField = setEmptyIfNull(
-      getObjectPropertyValueFromStringPath(
-        xDeal,
-        mapDataModel(xDeal, sortBy.field),
-      ),
-    );
-
-    const yField = setEmptyIfNull(
-      getObjectPropertyValueFromStringPath(
-        yDeal,
-        mapDataModel(yDeal, sortBy.field),
-      ),
-    );
-
-    if (sortBy.order === CONSTANTS.DEALS.SORT_BY.ASCENDING) {
-      if (xField > yField) {
-        return 1;
-      }
-
-      if (yField > xField) {
-        return -1;
-      }
-    }
-
-    if (sortBy.order === CONSTANTS.DEALS.SORT_BY.DESCENDING) {
-      if (xField > yField) {
-        return -1;
-      }
+const findDeals = async (queryParameters) => {
+  const dealsCollection = await db.getCollection('tfm-deals');
 
-      if (yField > xField) {
-        return 1;
-      }
-    }
+  const {
+    searchString, sortBy, fieldQueries, pagesize, page = 0
+  } = queryParameters;
 
-    return 0;
-  });
+  let query = {};
 
-const findDeals = async (searchString, sortBy, fieldQueries, callback) => {
-  const dealsCollection = await db.getCollection('tfm-deals');
+  const selectedField = {};
+  if (sortBy) {
+    sortBy.bssField = getBSSProperty(sortBy.field);
```

Comment:
I'm a little confused but should this be selectedField.bssField?
If so could we call selectedField, `fieldToSortOn` or something

Reply:
Can see from later it's not, I wonder if we take out the bssField and field properties into a bssField and a nonBssField variables to make it clear rather than changing the input sortBy parameter